### DISCUSSION TITLE
Is the 5.2.0-beta1 virtual threads compatible?

### DISCUSSION THREAD

**Discussion by joanbalaguero** [xetorthio/jedis#3879] [Q&A]
Hello,

Is the 5.2.0-beta1 virtual threads compatible? Basically by replacing sync regions by Reentrant locks. We are moving to jdk21, if 5.2.0-beta1 is VT compatible we could test it intensively.

Thanks,

Joan.

**Comment by sazzad16** [comment]
No.

**Reply by joanbalaguero** [reply]
Thanks for the quick response. And is it planned to be?

**Reply by sazzad16** [reply]
That's under consideration for quite a long time now. But neither accepted nor rejected yet.

**Comment by sazzad16** [comment]
#3480 was merged and should part of 5.2.0 (final release).
The change is available in latest 5.2.0-SNAPSHOT but may OR may not be part of any alpha/beta release.

**Reply by sazzad16** [reply]
UPDATE: commons-pool2 library is required an update. [Tracking issue](https://github.com/redis/jedis/issues/3927)
